### DISCUSSION TITLE
Import CSV with different datetime formats

### DISCUSSION THREAD

**Discussion by hmilbi** [duckdb/duckdb#9685] [Q&A] [answered]
Hi,
unfortunately I have to deal with csv files that have multiple different dateTime formats in a single row.

"11/14/2023 9:04:37 PM"
"2016-12-15T00:00:00.0000000"
"1900-01-01T00:00:00.0000000Z"
"1900-01-01T00:00:00.0000000+00:00"

At least each column is consistent in its format.

Currently I am importing these as string and cast them in the select statement. 

Is there any other way to solve this and have this automatically imported into timestamp or dateTime columns?

**Comment by Tishj** [comment]
<https://duckdb.org/docs/data/csv/overview>

There exists a `timestampformat` option, not sure if that's what you're looking for?

**Reply by hmilbi** [reply]
Thank you for your reply. I already tried that. The problem is that I can only provide one format but I have four different kinds

**Comment by l1t1** [comment]
>Currently I am importing these as string and cast them in the select statement.

you can create a new table based on above select cast ...
then query the new table

**Reply by hmilbi** [reply]
Yeah this is what I am doing right now. Hoping to find an alternative approach. Looking right now into pandas and polars as an intermediate step

**Comment by hmilbi** [comment, marked as answer]
The csv parser of Polars is able to load my csv's so that I do not have to know the format in advance
